### PR TITLE
Update AWS SDK version to 1.10.57

### LINKED ISSUE #5123: Failure on running the script setup-adapters.sh
### Problem description

I am facing an issue when trying to run the script setup-adapters.sh to download the [AWS SDK C++](https://github.com/aws/aws-sdk-cpp) library.

### System information

Velox System Info v0.0.2
Commit: cc1cec16d80917c1228191eab16a1e5c9ced958b
CMake Version: 3.26.4
System: Darwin-22.4.0
Arch: arm64
C++ Compiler: /Library/Developer/CommandLineTools/usr/bin/c++
C++ Compiler Version: 14.0.3.14030022
C Compiler: /Library/Developer/CommandLineTools/usr/bin/cc
C Compiler Version: 14.0.3.14030022
CMake Prefix Path: /Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr;/opt/homebrew;/usr/local;/usr;/;/opt/homebrew/Cellar/cmake/3.26.4;/usr/local;/usr/X11R6;/usr/pkg;/opt;/sw;/opt/local
  

### CMake log

```bash
[2/272] Building CXX object crt/aws-crt-cpp/CMakeFiles/aws-crt-cpp.dir/source/external/cJSON.cpp.o
FAILED: crt/aws-crt-cpp/CMakeFiles/aws-crt-cpp.dir/source/external/cJSON.cpp.o
/Library/Developer/CommandLineTools/usr/bin/c++ -DAWS_MQTT_WITH_WEBSOCKETS -DAWS_USE_KQUEUE -DCJSON_HIDE_SYMBOLS -DDEBUG_BUILD -DENABLE_COMMONCRYPTO_ENCRYPTION -DENABLE_CURL_CLIENT -DPLATFORM_APPLE -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-http/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-io/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-common/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/_build/crt/aws-crt-cpp/crt/aws-c-common/generated/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-cal/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-compression/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-mqtt/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-auth/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-checksums/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-event-stream/include -I/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/crt/aws-c-s3/include -mcpu=apple-m1+crc -std=c++17 -fvisibility=hidden -g -std=gnu++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk -fPIC -Wall -Wno-long-long -pedantic -Werror -MD -MT crt/aws-crt-cpp/CMakeFiles/aws-crt-cpp.dir/source/external/cJSON.cpp.o -MF crt/aws-crt-cpp/CMakeFiles/aws-crt-cpp.dir/source/external/cJSON.cpp.o.d -o crt/aws-crt-cpp/CMakeFiles/aws-crt-cpp.dir/source/external/cJSON.cpp.o -c /Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/source/external/cJSON.cpp
/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/source/external/cJSON.cpp:146:5: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
    sprintf(version, "%i.%i.%i", CJSON_VERSION_MAJOR, CJSON_VERSION_MINOR, CJSON_VERSION_PATCH);
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/source/external/cJSON.cpp:584:18: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
        length = sprintf((char*)number_buffer, "null");
                 ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/source/external/cJSON.cpp:589:18: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
        length = sprintf((char*)number_buffer, "%1.15g", d);
                 ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/source/external/cJSON.cpp:595:22: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
            length = sprintf((char*)number_buffer, "%1.17g", d);
                     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/Users/imjalpreet/Documents/OSS/presto/presto-native-execution/adapter-deps/aws-sdk-cpp/crt/aws-crt-cpp/source/external/cJSON.cpp:1028:21: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
                    sprintf((char*)output_pointer, "u%04x", *input_pointer);
                    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
5 errors generated.
[13/272] Building CXX object crt/aws-crt-cpp/CMakeFiles/aws-crt-cpp.dir/source/http/HttpConnection.cpp.o
ninja: build stopped: subcommand failed.
```

### PR DESCRIPTION
**Pull request by imjalpreet**
Update the default version of AWS SDK in the setup script to 1.10.57

Resolves #5123

### CODE REVIEW

**Comment by majetideepak** [comment]
@imjalpreet CI is failing. I believe the newer versions of AWS SDK are not creating the `bin` folder. You need to manually create it in `.circleci/config.yml` for the `linux-adapters` run.
`mkdir ~/adapter-deps/install/bin/`

**Comment by imjalpreet** [comment]
@majetideepak Thanks, made the change to create the directory manually.

**Comment by majetideepak** [comment]
@kgpai can you please help merge this? Thanks.

**Comment by majetideepak** [comment]
@imjalpreet I see a S3 test failing. Do you want to investigate?

File: .circleci/config.yml [hunk 1]
```diff
@@ -430,7 +430,7 @@ jobs:
       - run:
           name: "Install Adapter Dependencies"
           command: |
-            mkdir ~/adapter-deps ~/adapter-deps/install
+            mkdir ~/adapter-deps ~/adapter-deps/install ~/adapter-deps/install/bin
```
Comment (kgpai) [inline .circleci/config.yml:433]: nit: you can just use mkdir -p ~/adapter-deps/install/bin

**Comment by kgpai** [comment]
Will merge after the s3 test is fixed. Thanks @imjalpreet .

**Review by majetideepak — APPROVED** [review]

**Comment by majetideepak** [comment]
Thanks, @imjalpreet

**Comment by majetideepak** [comment]
@kgpai can you help merge this? Thanks.

**Comment by majetideepak** [comment]
@imjalpreet Can you rebase with main branch?

File: .circleci/dist_compile.yml [hunk 2]
```diff
@@ -391,7 +391,7 @@ jobs:
       - run:
           name: "Install Adapter Dependencies"
           command: |
-            mkdir ~/adapter-deps ~/adapter-deps/install
```
Comment (kgpai) [inline .circleci/dist_compile.yml:394]: Please make this change back in .config/circleci.yml. ( Apologies, for reverting this ).

Comment (imjalpreet) [reply .circleci/dist_compile.yml]: @kgpai I reverted it back to config.yml.

**Review by majetideepak — APPROVED** [review]

**Pull request closed without merging**